### ISSUE TITLE
new version 4_1

### ISSUE THREAD

**Issue by patience4711** [patience4711/RPI-APS-inverters#121]
The kernel on the version 3_x is several years old, Sadly it is not possible to issue software updates that also upgrades the linux verson. So i rebuild the system on a new linux release (bookworm). This involved a lot of work and testing because a lot of things work differently in this new version, For example all scripts that use the communication port had to be revised.  I also took the opportunity to improve some of the scripts and html.

From now on the old versions are not updated anymore because i have no working system with 3_9 so i can't test anything. This doesn't mean that your current system is useless now, when it works you can enjoy it for years.

However if you want to be up-to-date you can start with the new system. The only way is to burn the image to a (new) sd-card, Remember that it is possible to backup and restore the database of your old system.

I have been using the new system for several months without any problems. 

Recently i published the sdcard image, version 4_2. If you are going to use it, please expand the filesystem first before you do anything else.

known issues: If you restore a database backup of a previous 3_x version, your system inherits the version number. Also the the info on sd-card usage may be incorrect.  This isn't a problem except that it causes confusion. These issues are solved in v4_2. 

I published an upgrade to version 4_2c that relocates the info files to prevent that they are overwritten by a database restore. Install it if you are running v4_0 or 4_1. After the upgrade you may see the expand button in the system page. You can re-expand the filesystem to get rid of it.

**Comment by autourdupc** [comment]
Hello Patience4711.

I will soon move to new version, and I thought about a new feature...

Are you aware of a backup function like the one we can find in Freepbx distro named "raspbx-backup"  : it backups the complete SD-CARD to an image file.
It works this way : 
1) First run it creates (while system is running in production) an image file of the entire SD-CARD (you pass settings to tell path and name - Can be elsewhere on a mapped drive or FTP...)
2) Each new run (while system is still running in production) updates the existing image file (I personnaly use a cron job to update remotely on an FTP outside server my Freepbx SD-CARD)
  -> The image file is exactly the mirrot of the SD-CARD running, so its easy to duplicate / create a new SD-CARD in case of corruption without any settings.

I think there is an open project doing this raspbx-backup function... It would be cool for all your projets.

Regards,
Laurent.

**Comment by patience4711** [comment]
Hi Laurent, this sounds good. There are however some concerns. 
First of all, the image should be saved to an external storage, otherwise it has no use. This means extra hardware and space. And when we have this external storage, it would be better to run the whole system from this medium.  But when this gets corrupted we w'd lose our image too. 

So basically it could work like this:
 - fit a usb stick (need an adapter cable microusb to usb or something)
 - install Asterisk for raspberry pi (no version for bookworm availlable and pi-zero not supported)
-  have a script that runs the backup program every night.

We also need to have a setting that the admin can use to enable/disable the backup. 

Saving the image via ftp is something for only advanced users. 

So this all seems not simply to implement at this moment.

**Comment by autourdupc** [comment]
Hello @patience4711 

I've just installed last 4.2 version bec. got an SD card corruption, then restored a previous 4.1 backup (and at this time I did an expand as per requested).

I notice now that boot process is very very long... It takes a while to get the ssh login, same for web index page.
It was not so long in 4.1 version (maybe 4.2a version), I do not really remember.

I found alos that CPU load is highter now 1.5 to 2.5 in Top command, while not poolling at night.

Did you change something that could increase system load ?

Regards,
Laurent.

**Comment by patience4711** [comment]
No, the changelog explains what i changed, nothing that could increase the cpu load.

**Comment by autourdupc** [comment]
Hello @patience4711 

I think I found some explanations...
Testing my Samsung Evo SD-Card, I found a R/W speed up to 4,5 MB using dd command... This is so poor !

So according to this page
https://bret.dk/best-raspberry-pi-micro-sd-cards/#Raspberry-Pi-Zero-2-microSD-Card-Performance
I ordered a kioxia SD-CARD and will see if it works better when I got it !
